### PR TITLE
refactor: bump opendal to v0.13.0

### LINKED ISSUE #7151: Bump OpenDAL to 0.13.0
OpenDAL has released 0.13.0: https://github.com/datafuselabs/opendal/discussions/538

In this version bump, we need to:

- [x] Bump versions
- [x] Add support for gcs as backend storage #7023
- [ ] Add support for gcs as stage #7172 
- [x] Enable feature `layer-retry` (replace deprecated `with_backoff`)
- [x] Enable feature `layer-tracing`
- [x] Enable feature `layer-metrics`
- [x] Enable feature `layer-logging`

### PR DESCRIPTION
**Pull request by ClSlaid**
Signed-off-by: ClSlaid <cailue@bupt.edu.cn>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR will bump the dependency of OpenDAL to v0.13.0

Part of #7151

### CODE REVIEW

**Review by ZeaLoVe — APPROVED** [review]

File: src/common/catalog/Cargo.toml [hunk 1]
```diff
@@ -27,5 +27,5 @@ common-users = { path = "../users" }
 
 async-trait = "0.1.56"
 dyn-clone = "1.0.6"
-opendal = { version = "0.12.0", features = ["retry"] }
+opendal = { version = "0.13.0", features = ["retry"] }
```
Comment (Xuanwo) [inline src/common/catalog/Cargo.toml:30]:
```suggestion
opendal = { version = "0.13.0", features = ["layers-retry"] }
```

File: src/common/config/Cargo.toml [hunk 2]
```diff
@@ -18,7 +18,7 @@ common-tracing = { path = "../tracing" }
 
 clap = { version = "3.2.5", features = ["derive", "env"] }
 once_cell = "1.12.0"
-opendal = { version = "0.12.0", features = ["retry", "compress"], optional = true }
+opendal = { version = "0.13.0", features = ["retry", "compress"], optional = true }
```
Comment (Xuanwo) [inline src/common/config/Cargo.toml:21]:
```suggestion
opendal = { version = "0.13.0", features = ["layers-retry", "compress"], optional = true }
```

File: src/query/service/src/sessions/session_mgr.rs [hunk 3]
```diff
@@ -410,9 +412,9 @@ impl SessionManager {
 
     // Init the storage operator by config.
     async fn init_storage_operator(conf: &Config) -> Result<Operator> {
-        let op = init_operator(&conf.storage.params)?;
         // Enable exponential backoff by default
-        let op = op.with_backoff(backon::ExponentialBackoff::default());
+        let op = init_operator(&conf.storage.params)?
+            .layer(RetryLayer::new(ExponentialBackoff::default()));
```
Comment (Xuanwo) [inline src/query/service/src/sessions/session_mgr.rs:417]: Please also enable `logging`, `tracing`, `metrics`.

File: src/common/contexts/Cargo.toml [hunk 4]
```diff
@@ -12,4 +12,4 @@ test = false
 common-base = { path = "../base" }
 
 async-trait = "0.1.56"
-opendal = { version = "0.12.0", features = ["retry"] }
+opendal = { version = "0.13.0", features = ["retry"] }
```
Comment (Xuanwo) [inline src/common/contexts/Cargo.toml:15]: ditto

File: src/common/pipeline/sources/Cargo.toml [hunk 5]
```diff
@@ -23,5 +23,5 @@ common-streams = { path = "../../streams" }
 async-trait = { git = "https://github.com/datafuse-extras/async-trait", rev = "f0b0fd5" }
 futures = "0.3.21"
 futures-util = "0.3.21"
-opendal = { version = "0.12.0", features = ["retry"] }
+opendal = { version = "0.13.0", features = ["retry", "compress"] }
```
Comment (Xuanwo) [inline src/common/pipeline/sources/Cargo.toml:26]: ditto

File: src/common/storage/Cargo.toml [hunk 6]
```diff
@@ -12,6 +12,6 @@ storage-hdfs = ["opendal/services-hdfs"]
 [dependencies]
 anyhow = "1.0.58"
 globiter = "0.1.0"
-opendal = { version = "0.12.0", features = ["retry", "services-http"] }
+opendal = { version = "0.13.0", features = ["retry", "services-http"] }
```
Comment (Xuanwo) [inline src/common/storage/Cargo.toml:15]: ditto

**Comment by Xuanwo** [comment]
`retry` feature has been deprecated. Please replace with `layers-retry`: 

https://github.com/datafuselabs/opendal/blob/da22720c50dd27cd7375438abc248bce6aad5835/Cargo.toml#L33-L34

File: src/query/service/src/sessions/session_mgr.rs (hunk 3 above)
Comment (drmingdrmer) [inline src/query/service/src/sessions/session_mgr.rs]: 👍

**Review by drmingdrmer — APPROVED** [review]
Looks good :)

File: src/query/service/src/sessions/session_mgr.rs [hunk 7]
```diff
@@ -410,9 +414,11 @@ impl SessionManager {
 
     // Init the storage operator by config.
     async fn init_storage_operator(conf: &Config) -> Result<Operator> {
-        let op = init_operator(&conf.storage.params)?;
         // Enable exponential backoff by default
-        let op = op.with_backoff(backon::ExponentialBackoff::default());
+        let op = init_operator(&conf.storage.params)?
+            .layer(RetryLayer::new(ExponentialBackoff::default()))
+            .layer(LoggingLayer)
+            .layer(MetricsLayer);
```
Comment (Xuanwo) [inline src/query/service/src/sessions/session_mgr.rs:421]: tracing...

Comment (ClSlaid) [reply src/query/service/src/sessions/session_mgr.rs:423]: sorry for that

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**